### PR TITLE
Make port configurable to allow integrations

### PR DESCRIPTION
**Pull request by npwalker**

### CODE REVIEW

**Review by reidmv — APPROVED** [review]
Looks good to me conceptually. If you've tested it and it works I'm all :+1:

**Comment by reidmv** [comment]
Oh; I was going to ask though, can a better description of what integrations this is interesting for and why we want it be put in the commit message or PR description? I still feel in the dark about what this actually enables, specifically.

**Comment by npwalker** [comment]
I'm trying to pipe the output from the metrics script, through json2graphite and over to a prometheus instance.  

The graphite conversion uses port 9109 by default.  

https://github.com/prometheus/graphite_exporter/

**Review by jarretlavallee — APPROVED** [review]
👍 Looks good to me.

**Comment by npwalker** [comment]
For reference, to pump metrics over to prometheus I had to:
1.  Change the output-dir of the the metric script to a new directory
2.  Have the pattern on json2graphite look in the new directory from above
3.  Clear the directory after json2graphite completes (so there's only ever 1 set of files in the dir)

Not ideal but workable.

**Comment by reidmv** [comment]
Interesting. Why didn't piping work?

```
tk_metrics --print --more-flags | json2graphite --flags
```

The files then would be for support only, and the stdout of the command would be for prometheus, avoiding the need for new directories or cleanup. That's the pattern we were aiming to enable. Is something preventing this from working correctly?

**Comment by npwalker** [comment]
1.  I don't know how to make json2graphite read STDIN.  
    1.  It looked to me like it only accepts files

2.  I tried redirecting the output from tk_metrics into a file ( tk_metrics > /tmp/file ) and then reading it but when multiple servers are queried the resulting JSON blob wasn't acceptable for json2graphite.  

I think even if 1 works then it should have the same problem as 2.

**Comment by reidmv** [comment]
Oh, right... the original version of the script needed the filename to extract date information. Newer versions have that info in the JSON itself, but the script wasn't written to accept that, it was adapted to later. https://github.com/npwalker/puppet-metrics-viewer/blob/73f5a5203ab8c2fbe76ee25ed7f7c6cc1ad4f23b/json2graphite.rb#L43

The script should be modifed to accept STDIN, whenever somebody has the time.

Sounds like it also needs to be fixed to accept multiple servers, if that's not working anymore. I thought it did at one point in time.

File: json2graphite.rb [hunk 1]
```diff
@@ -153,8 +153,13 @@ def metrics(data, timestamp, parent_key = nil)
 OptionParser.new do |opt|
   opt.on('--pattern PATTERN') { |o| $options[:pattern] = o }
   opt.on('--netcat HOST') { |o| $options[:host] = o }
+  opt.on('--port PORT') { |p| $options[:port] = p }
 end.parse!
 
+if $options[:port].nil?
```
Comment (npwalker) [inline json2graphite.rb]: This doesn't actually seem to work.  I only tested passing in a new port previously.  Now I tested trying to run view-in-graphana.sh and it doesn't work.

**Comment by npwalker** [comment]
Closing in favor of https://github.com/puppetlabs/puppet-metrics-viewer/pull/22 since the influxdb changes obsoleted this PR.

**Pull request closed without merging**